### DISCUSSION TITLE
Quarto journal extension - how to reference the local `.cls` file?

### DISCUSSION THREAD

**Discussion by pat-alt** [quarto-dev/quarto-cli#4046] [Q&A] [answered]
I've recently started working on some Custom Format extensions (e.g. this simple one [here](https://github.com/pat-alt/quarto-julia) for Julia-themed HTML). Now I was trying to step it up a notch and create a Journal Article Extension (this [repo](https://github.com/pat-alt/quarto-juliacon-proceedings/tree/main/_extensions/juliacon-proceedings)) for JuliaCon Proceedings derived off the existing [template repository](https://github.com/JuliaCon/JuliaConSubmission.jl). From there I've lifted the relevant what I think should be the relevant `.tex`, `.cls`, ... files and then tried to follow the [guide](https://quarto.org/docs/journals/formats.html#example-acm-format) for creating Journal Articles. 

I'm trying to use an article template that uses the local class file [`juliacon.cls`](https://github.com/pat-alt/quarto-juliacon-proceedings/blob/main/_extensions/juliacon-proceedings/juliacon.cls) like so:

```{tex}
% JuliaCon proceedings template
\documentclass{juliacon}
\setcounter{page}{1}
\begin{document}
$body$
\end{document}
```

When I then try to render the [`template.qmd`](https://github.com/pat-alt/quarto-juliacon-proceedings/blob/main/template.qmd) file, I get the following error:

```
updating tlmgr

updating existing packages
finding package for juliacon.cls

compilation failed- no matching packages
LaTeX Error: File `juliacon.cls' not found.
```

What am I doing wrong? Any help would be much appreciated 🙏🏽

**Comment by dragonstyle** [comment, marked as answer]
I'd recommend that you copy the `cls` file to the input directory as a part of rendering (the latex compilers will generally look in the document directory to resolve such dependencies). A good example of this is here:

https://github.com/quarto-journals/elsevier/tree/main/_extensions/elsevier

You can see that the class file is included in the extension and is listed in `format-resources`, which means that it be copied to the input directory as a part of rendering.

Appreciate your effort here and hope this helps!

**Reply by pat-alt** [reply]
Thanks very much @dragonstyle , that did it!

**Comment by fpatz** [comment]
Looking into the same issue, I've found this in `_extension.yml` to work:

```
title: A LaTeX format extension
contributes:
  format:
    pdf:
      latex-input-paths:
        - _extensions/path-to-my-extension
```

No idea, whether this is a robust solution, though ...

**Reply by cderv** [reply]
@dragonstyle what is your insight about setting this in the extension ?

**Reply by fpatz** [reply]
It works for my case, but I have no idea whether it will work when you have `latex-input-paths` in multiple extensions or in the document file. I'd expect quarto to merge the path lists, but haven't checked.

**Reply by dragonstyle** [reply]
I expect that will work just fine. 

We recommend using the `format-resources` to copy the `cls` file to the input's working directory mostly because this simplifies debugging LaTeX errors that you may encounter as you work. By copying the `cls` to the input directory, you can use options like `keep-tex` to preserve the LaTeX output and then just use normal commands like `xelatex` or `pdflatex` to compile the LaTeX to a pdf manually. If you use `latex-input-paths`, you'll need to provide that path to the `xelatex` or `pdflatex` command as well, which makes debugging a little more annoying.
